test/extractor/ConfigurationExtractor.test.ts

```diff
@@ -59,6 +59,41 @@ describe('ConfigurationExtractor', () => {
       expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
     });
 
+    it('should be able to handle a specified FHIR version', () => {
+      const ig = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'simple-ig.json'), 'utf-8')
+      );
+      const result = ConfigurationExtractor.process([ig, ...resources], '4.5.0');
+      expect(result.config.fhirVersion).toEqual(['4.5.0']);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(0);
+    });
+
+    it('should be able to handle a specified FHIR version when FHIR version is missing', () => {
+      const ig = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'missing-fhir-version-ig.json'), 'utf-8')
+      );
+      const result = ConfigurationExtractor.process([ig, ...resources], '5.0.0');
+      expect(result.config.fhirVersion).toEqual(['5.0.0']);
+      expect(loggerSpy.getAllMessages('warn')).toHaveLength(2);
+      expect(loggerSpy.getMessageAtIndex(-2, 'warn')).toMatch(
+        'ImplementationGuide missing properties needed to generate configuration file: fhirVersion. These properties will be inferred based on FHIR definitions.'
+      );
+      expect(loggerSpy.getLastMessage('warn')).toMatch(
+        'FHIR Version mismatch error: specified version is: 5.0.0 while detected version is: 4.0.1. GoFSH will use the specified version (5.0.0), but this may result in additional processing errors.'
+      );
+    });
+
+    it('should log an error when the specified FHIR version is different from detected FHIR version', () => {
+      const ig = JSON.parse(
+        fs.readFileSync(path.join(__dirname, 'fixtures', 'simple-ig.json'), 'utf-8')
+      );
+      const result = ConfigurationExtractor.process([ig, ...resources], '5.0.0');
+      expect(result.config.fhirVersion).toEqual(['5.0.0']);
+      expect(loggerSpy.getLastMessage('warn')).toMatch(
+        'FHIR Version mismatch error: specified version is: 5.0.0 while detected version is: 4.5.0. GoFSH will use the specified version (5.0.0), but this may result in additional processing errors.'
+      );
+    });
```

Comment:
I think it might be good to include one more test here to cover the case when a FHIR Version is not detected and not specified so it defaults to `4.0.1`. Does that make sense to you to include? Or do you feel like that's covered elsewhere?

Reply:
I like that.

Reply:
I believe this was covered in the `should be able to handle a specified FHIR version when FHIR version is missing` test case in `ConfigurationExtractor.test.ts`. Let me know if we can get more specific.

Reply:
I think that case covers when a FHIR version isn't present in the resources but it is specified directly to the `ConfigurationExtractor.process` method. I was thinking of one that used those same resources that are missing a FHIR version but there is no user provided FHIR version either. Something like this:

```
const ig = JSON.parse(
  fs.readFileSync(path.join(__dirname, 'fixtures', 'missing-fhir-version-ig.json'), 'utf-8')
);
const result = ConfigurationExtractor.process([ig, ...resources]);
expect(result.config.fhirVersion).toEqual(['4.0.1']);
// ... other expects for the appropriate warnings
```